server/src/test/java/org/elasticsearch/index/query/MatchIntervalsSourceProviderTests.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.query.IntervalsSourceProvider.Match;
+
+public class MatchIntervalsSourceProviderTests extends AbstractSerializingTestCase<Match> {
+
+    @Override
+    protected Match createTestInstance() {
+        return IntervalQueryBuilderTests.createRandomMatch(0, randomBoolean());
+    }
+
+    @Override
+    protected Match mutateInstance(Match instance) throws IOException {
+        String query = instance.getQuery();
+        int maxGaps = instance.getMaxGaps();
+        boolean isOrdered = instance.isOrdered();
+        String analyzer = instance.getAnalyzer();
+        IntervalsSourceProvider.IntervalFilter filter = instance.getFilter();
+        String useField = instance.getUseField();
+        switch (between(0, 5)) {
+            case 0:
+                query = randomAlphaOfLength(3);
```

Comment:
Can this clash with the original query or does the length prevent this? Maybe add a short note if so.

Reply:
You are right, it could clash with the original query. Fixed.